docs/src/locators.md

```diff
@@ -578,60 +603,133 @@ await page.Locator("#tsf > div:nth-child(2) > div.A8SBwf > div.RNNXgb > div > di
+<div data-testid='product-card'>
+  <span>Product 2</span>
+  <button>Buy</button>
+</div>
+```
+
+You can click on the second product card by first locating the product cards by test id and then filtering to find the product card with the text of "Product 2".
 
 ```js
-await page.getByTestId('product-card').filter({ hasText: 'Product 3' }).click();
-await page.getByTestId('product-card').filter({ hasText: /product 3/ }).click();
+await page.getByTestId('product-card')
+    .filter({ hasText: 'Product 2' })
+    .click();
+
+await page.getByTestId('product-card')
+    .filter({ hasText: /product 2/ })
+    .click();
 ```
+
 ```java
-page.getByTestId("product-card").filter(new Locator.FilterOptions().setHasText("Product 3")).click();
-page.getByTestId("product-card").filter(new Locator.FilterOptions().setHasText(Pattern.compile("Product 3"))).click();
+page.getByTestId("product-card")
+    .filter(new Locator.FilterOptions().setHasText("Product 2"))
+    .click();
+    
+page.getByTestId("product-card")
+    .filter(new Locator.FilterOptions().setHasText(Pattern.compile("Product 2")))
+    .click();
 ```
+
 ```python async
-await page.get_by_test_id("product-card").filter(has_text="Product 3").click()
-await page.get_by_test_id("product-card").filter(has_text=re.compile("Product 3")).click()
+await page.get_by_test_id("product-card")
+    .filter(has_text="Product 2")
+    .click()
+
+await page.get_by_test_id("product-card")
+    .filter(has_text=re.compile("Product 2"))
+    .click()
 ```
+
 ```python sync
-page.get_by_test_id("product-card").filter(has_text="Product 3").click()
-page.get_by_test_id("product-card").filter(has_text=re.compile("Product 3")).click()
+page.get_by_test_id("product-card")
+    .filter(has_text="Product 2")
+    .click()
+
+page.get_by_test_id("product-card")
+    .filter(has_text=re.compile("Product 2"))
+    .click()
 ```
+
 ```csharp
-await page.GetByTestId("product-card").Filter(new() { HasText = "Product 3" }).ClickAsync();
-await page.GetByTestId("product-card").Filter(new() { HasText = new Regex("Product 3") }).ClickAsync();
+await page.GetByTestId("product-card")
+    .Filter(new() { HasText = "Product 2" })
+    .ClickAsync();
+
+await page.GetByTestId("product-card")
+    .Filter(new() { HasText = new Regex("Product 2") })
+    .ClickAsync();
 ```
 
-#### Filter by another locator
+### Filter by another locator
 
 Locators support an option to only select elements that have a descendant matching another locator.
 
+For example, consider the following DOM structure:
+
+```html
+<header>
+  <button>Sign in</button>
+</header>
+
+<section>
+  <button>Sign in</button>
+</section>
+```
+To select the "Sign in" button from the `section` we can first locate by the role of section and then filter by the button role.
+
 ```js
-page.getByRole('section').filter({ has: page.getByTestId('subscribe-button') })
+await page.getByRole('section')
+    .filter({ has: page.getByRole('button', { name: 'Sign in' })
+    .click()
```

Comment:
I don't think clicking on the section makes sense. Perhaps `toHaveText` is better?

Reply:
i thought this was clicking on the button. as it filters the button in the section or am i missing something??